### ISSUE TITLE
Side effects when override findBy and findOneBy methods with query builder

### ISSUE THREAD

**Issue by emr** [doctrine/orm#7833]
Actually, this side effect is not related to overriding`findBy` and `findOneBy` methods, it's related to using query builder.

I've overridden `findBy` and `findOneBy` methods to adding some query hints to all my `find*()` calls. But since I've done this, some side effects that I don't want to happen have occurred.

One case of them that I noticed in my Symfony project:

Simplified code flow:
```php
// fetch an entity
$object = $repository->find(1);
// make some updates on object (what I do is handling a form data on the object)
$object->setSlug('abc');
// and another request to repository for another purpose (what I do is uniqueness check)
$result = $repository->findBy(['slug' => 'abc']);
# some logic to determinate is the slug unique
```

|    Q        |   A
|------------ | -----
| Version     | x.y.z

**Comment by SenseException** [comment]
Hi @emr, it would be helpful if you also put a code example of your overridden `findby()` method into this issue. Reading and possibly reproducing the described behavior should make it easier to help.

**Comment by emr** [comment]
@SenseException I've added the code block to the issue.

**Comment by Ocramius** [comment]
Hmm, DQL queries don't implicitly refresh `UnitOfWork` entries: are you setting anh DQL hints on that query?

**Comment by emr** [comment]
@Ocramius @Ocramius I've updated the code block to show explicitly all my hints I adding to queries.

**Comment by emr** [comment]
@Ocramius I've updated the code block to show explicitly all the hints I've added to the queries.

**Comment by lcobucci** [comment]
@emr I think this is related to how `Gedmo\Translatable` works and the magic around it. Perhaps this can be avoided if you don't hydrate the entity (by using `EntityRepository#count()`, for example).

**Comment by emr** [comment]
@lcobucci of course, if I don't hydrate the entity, so any entity never gonna load to the memory.
